src/Xamarin.Android.Build.Tasks/Utilities/MarshalMethodsAssemblyRewriter.cs

```diff
@@ -53,11 +53,11 @@ public void Rewrite (bool brokenExceptionTransitions)
 			TypeDefinition androidEnvironment = FindType (monoAndroidRuntime, "Android.Runtime.AndroidEnvironmentInternal", required: true)!;
 			MethodDefinition unhandledExceptionMethod = FindMethod (androidEnvironment, "UnhandledException", required: true)!;
 
-			TypeDefinition runtimeNativeMethods = FindType (monoAndroidRuntime, "Android.Runtime.RuntimeNativeMethods", required: true);
-			MethodDefinition monoUnhandledExceptionMethod = FindMethod (runtimeNativeMethods, "monodroid_debugger_unhandled_exception", required: true);
+			TypeDefinition runtimeNativeMethods = FindType (monoAndroidRuntime, "Android.Runtime.RuntimeNativeMethods", required: true)!;
+			MethodDefinition monoUnhandledExceptionMethod = FindMethod (runtimeNativeMethods, "monodroid_debugger_unhandled_exception", required: true)!;
 
-			AssemblyDefinition corlib = resolver.Resolve ("System.Private.CoreLib");
-			TypeDefinition systemException = FindType (corlib, "System.Exception", required: true);
+			AssemblyDefinition? corlib = resolver.Resolve ("System.Private.CoreLib");
+			TypeDefinition systemException = FindType (corlib!, "System.Exception", required: true)!;
```

Comment:
Check if `asm` is null and throw an appropriate `ArgumentNullException`. Do this instead of using `!`.